document/CoreFormat.mdk

```diff
@@ -735,11 +735,69 @@ and so on, with the last element referring to the most deeply nested construct.
 
 ### Constraints { #location-Constraints }
 
-Within any one location object, if the `analysisTarget` property (&sect;[#location-analysisTarget])
-and the `resultFile` property (&sect;[#location-resultFile]) would both refer to the same location,
-then the `analysisTarget` property shall be present and the `resultFile` property shall be absent.
-If the `analysisTarget` property and the `resultFile` property would refer to different locations,
-then the `analysisTarget` property and the `resultFile` property shall both be present.
+Depending on the information available to the tool that produces the SARIF log file,
+either or both of the `analysisTarget` property (&sect;[#location-analysisTarget]) and the
+`resultFile` property (&sect;[#location-resultFile]) shall be present.
+
+If the tool that produces the log file knows the analysis target, then the `analysisTarget`
+property shall be present.
+If the tool knows that the result file is different from the analysis target,
+then the `resultFile` property shall be present;
+otherwise the `resultFile` property shall be absent.
+
+~ IsoNote
+Generally, an analysis tool will know both the file it was instructed to scan
+(the analysis target) and the file in which it detects a problem (the result file).
+~
+
+~ IsoNumberedExample { @IsoNumberedExample=1 }
+Suppose a static analysis tool for C++ source code is instructed to scan the source file _a.cpp_,
+and suppose the tool detects a problem in _a.cpp_.
+In this case, the tool should set the `analysisTarget` property to `a.cpp`,
+and it should not set the `resultFile` property.
+~
+
+~ IsoNumberedExample
+Suppose a static analysis tool for C++ source code is instructed to scan the source file _a.cpp_,
+which includes the header file _b.h_,
+and suppose the tool detects a problem in _b.h_.
+In this case, the tool should set the `analysisTarget` property to `a.cpp`,
+and it should set the `resultFile` property to `b.h`.
+~
+
+~ IsoNumberedExample
+Suppose a static analysis tool for object code detects a problem in the binary file _c.dll_,
+and suppose the tool has available symbol information which maps that location within the binary
+to a specific line in a source file _d.cpp_.
+In this case, the tool should set the `analysisTarget` property to `c.dll`,
+and it should set the `resultFile` property to `d.cpp`.
+~
+
+If the tool that produces the log file does not know the analysis target,
+then the `resultFile` property shall be present and the `analysisTarget` property shall be absent.
+
+~ IsoNote
+Some static analysis tools produce output in a format that does not include both the
+analysis target and the result file.
+In such cases, a conversion tool which translates the output into the SARIF format
+might only have the result file available. 
+~
+
+~ IsoExample
+Suppose a static analysis tool for C++ source code is instructed to scan the source file _a.cpp_,
```

Comment:
This example would be most useful if we explicitly noted that we don't know how the scanner was configured. That's the scenario in which a converter would use resultFile (assuming that there was in fact some ambiguity about the scanning tool behavior. i.e., if it were true that the scanner might have either been configured to scan the .h explicitly or if it were possible it had reported against it due to scanning a .c file that included the header).